### PR TITLE
fix: added constructors required by Uno.UI for Android and iOS.

### PR DESCRIPTION
**Pull request by weitzhandler**
**What kind of change does this PR introduce?**
Fixes missing constructors by Uno.UI.

**What is the current behavior?**
Uno complains about these missing constructors.

**What is the new behavior?**
Adds bridge constructors in the intermediate classes.

**What might this PR break?**
None

**Please check if the PR fulfills these requirements**
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

@jeromelaban Please have a look

### CODE REVIEW

**Comment by glennawatson** [comment]
https://github.com/reactiveui/ReactiveUI/blob/master/src/Directory.build.targets use the constants in that file rather than the others.

File: src/ReactiveUI/Platforms/windows-common/ReactivePage.cs [hunk 1]
```diff
@@ -100,6 +104,38 @@ class ReactivePage<TViewModel> :
                 typeof(ReactivePage<TViewModel>),
                 new PropertyMetadata(null));
 
+#if __ANDROID__
```
Comment (glennawatson) [inline src/ReactiveUI/Platforms/windows-common/ReactivePage.cs]: Use the one in that file

File: src/ReactiveUI/Platforms/windows-common/ReactivePage.cs [hunk 2]
```diff
@@ -100,6 +104,38 @@ class ReactivePage<TViewModel> :
...
+        /// objects that may have been collected in the managed world.
+        /// </remarks>
+        /// <param name="javaReference">javaReference.</param>
+        /// <param name="transfer">transfer.</param>
+        protected ReactivePage(IntPtr javaReference, global::Android.Runtime.JniHandleOwnership transfer)
+            : base(javaReference, transfer)
+        {
+        }
+#endif
+#if __IOS__
```
Comment (glennawatson) [inline src/ReactiveUI/Platforms/windows-common/ReactivePage.cs]: Use the one in that file

File: src/ReactiveUI/Platforms/windows-common/ReactivePage.cs [hunk 3]
```diff
@@ -100,6 +104,40 @@ class ReactivePage<TViewModel> :
...
+#if ANDROID
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ReactivePage{TViewModel}"/> class.
+        /// Native constructor, do not use explicitly.
+        /// </summary>
+        /// <remarks>
+        /// Used by the Xamarin Runtime to materialize native
+        /// objects that may have been collected in the managed world.
+        /// </remarks>
+        /// <param name="javaReference">javaReference.</param>
```
Comment (glennawatson) [inline src/ReactiveUI/Platforms/windows-common/ReactivePage.cs]: Put a valid comment in the Params

File: src/ReactiveUI/Platforms/windows-common/ReactiveUserControl.cs [hunk 4]
```diff
@@ -100,6 +104,40 @@ class ReactiveUserControl<TViewModel> :
...
+#if ANDROID
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ReactiveUserControl{TViewModel}"/> class.
+        /// Native constructor, do not use explicitly.
+        /// </summary>
+        /// <remarks>
+        /// Used by the Xamarin Runtime to materialize native
+        /// objects that may have been collected in the managed world.
+        /// </remarks>
+        /// <param name="javaReference">javaReference.</param>
```
Comment (glennawatson) [inline src/ReactiveUI/Platforms/windows-common/ReactiveUserControl.cs]: Put valid comments in the Params

File: src/ReactiveUI/Platforms/windows-common/ReactivePage.cs (hunk 3 above)
Comment (weitzhandler) [reply src/ReactiveUI/Platforms/windows-common/ReactivePage.cs]: TBH, I don't know myself.
I tried to look up the docs from [here](https://github.com/unoplatform/uno/blob/master/src/Uno.UI/UI/Xaml/Controls/Page/NativePage.Android.cs#L14) but it says nothing.

File: src/ReactiveUI/Platforms/windows-common/ReactiveUserControl.cs (hunk 4 above)
Comment (weitzhandler) [reply src/ReactiveUI/Platforms/windows-common/ReactiveUserControl.cs]: Same as [here](https://github.com/reactiveui/ReactiveUI/pull/2116#discussion_r301410650).

File: src/ReactiveUI/Platforms/windows-common/ReactivePage.cs [hunk 5]
```diff
@@ -100,6 +104,40 @@ class ReactivePage<TViewModel> :
...
+        /// <param name="transfer">transfer.</param>
```
Comment (glennawatson) [inline src/ReactiveUI/Platforms/windows-common/ReactivePage.cs]: The JniHandleOwnership is mentioned here https://docs.microsoft.com/en-us/xamarin/android/platform/java-integration/working-with-jni and https://docs.microsoft.com/en-us/dotnet/api/android.runtime.jnihandleownership?view=xamarin-android-sdk-9 -- basically it indicates how the JNI system would transfer ownership of the pointer behind the scenes.

The javaReference would be a pointer to the native form instance.

File: src/ReactiveUI/Platforms/windows-common/ReactivePage.cs [hunk 6]
```diff
@@ -100,6 +104,40 @@ class ReactivePage<TViewModel> :
...
+            : base(javaReference, transfer)
+        {
+        }
+#endif
+#if IOS
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ReactivePage{TViewModel}"/> class.
+        /// Native constructor, do not use explicitly.
+        /// </summary>
+        /// <param name="handle">handle.</param>
```
Comment (glennawatson) [inline src/ReactiveUI/Platforms/windows-common/ReactivePage.cs]: This would be a "Handle to the native control"

**Comment by glennawatson** [comment]
Cool thanks, looking good now.

File: src/ReactiveUI/Platforms/windows-common/ReactivePage.cs (hunk 5 above)
Comment (weitzhandler) [reply src/ReactiveUI/Platforms/windows-common/ReactivePage.cs]: Copied from [here](https://docs.microsoft.com/en-us/dotnet/api/java.lang.object.-ctor?view=xamarin-android-sdk-9#Java_Lang_Object__ctor_System_IntPtr_Android_Runtime_JniHandleOwnership_).

**Pull request merged**